core/src/main/java/org/hisp/dhis/android/core/sms/domain/interactor/ConfigCase.kt

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * Set the gateway number to send the SMS. This is a required parameter before sending SMS.
+     * @param gatewayNumber The gateway numberf
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setGatewayNumber(gatewayNumber: String?): Completable {
+        return if (gatewayNumber.isNullOrEmpty()) {
+            Completable.error(IllegalArgumentException("Gateway number can't be empty"))
+        } else localDbRepository.setGatewayNumber(gatewayNumber)
+    }
+
+    /**
+     * Delete the gateway number to send the SMS.
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun deleteGatewayNumber(): Completable {
+        return localDbRepository.deleteGatewayNumber()
+    }
+
+    /**
+     * Set if SMS Module is enabled or not. It is required to enable it before using it.
+     * @param enabled If the module is enabled or not
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setModuleEnabled(enabled: Boolean): Completable {
+        return localDbRepository.setModuleEnabled(enabled)
+    }
+
+    /**
+     * Set a new MetadataDownload configuration.
+     * @param metadataIdsConfig Configuration with the metadata ids.
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setMetadataDownloadConfig(metadataIdsConfig: GetMetadataIdsConfig?): Completable {
+        return if (metadataIdsConfig == null) {
+            Completable.error(IllegalArgumentException("Received null config"))
+        } else localDbRepository.setMetadataDownloadConfig(metadataIdsConfig)
+    }
+
+    val metadataDownloadConfig: Single<GetMetadataIdsConfig>
+        /**
+         * Get the metadata download configuration.
+         * @return `Single with the` metadata download configuration.
+         */
+        get() = localDbRepository.metadataDownloadConfig
+
+    /**
+     * Method to download metadata ids. This is required before using the SMS module.
+     */
+    suspend fun refreshMetadataIds() {
+
+        val enabled = localDbRepository.isModuleEnabled.await()
+
+        if (enabled) {
+            try {
+                val config = metadataDownloadConfig.await()
+                val metadata = webApiRepository.getMetadataIds(config).await()
+                localDbRepository.setMetadataIds(metadata).await()
+            } catch (ignored: Exception) {
+                Log.d(TAG, "Can't read saved SMS metadata download config. Using default.")
+                defaultMetadataDownloadConfig
+            }
+        } else {
+            Log.d(TAG, "Not refreshing SMS metadata, because sms module is disabled")
+        }
+
+    }
+
+    /**
+     * Callable that triggers the method [.refreshMetadataIds].
+     * @return Callable object to refresh metadata ids.
+     */
+    suspend fun refreshMetadataIdsCallable() {
```

Comment:
This is part of the internal API, we can use `suspend` here and modify the method with `internal`.